pkg/globalnet/controllers/ipam/gatewaymonitor.go

```diff
@@ -220,7 +220,30 @@ func (i *GatewayMonitor) initializeIpamController() {
 	}
 
 	klog.V(4).Infof("On Gateway Node, initializing ipamController.")
-	ipamController, err := NewController(i.ipamSpec, &informerConfig)
+	clusterList, err := i.submarinerClientSet.SubmarinerV1().Clusters(i.ipamSpec.Namespace).List(metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Error while retrieving the cluster info: %v", err)
+		return
+	}
+
+	globalCIDR := ""
+	for _, cluster := range clusterList.Items {
+		if cluster.Spec.ClusterID == i.clusterID {
+			klog.V(4).Infof("GlobalCidr configured in %s is %s", i.clusterID, cluster.Spec.GlobalCIDR)
+			if cluster.Spec.GlobalCIDR != nil {
```

Comment:
I think we should check `len(cluster.Spec.GlobalCIDR) > 0` to also handle an empty array.